protocol/java/org/openyolo/protocol/internal/IntentProtocolBufferExtractor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.protocol.internal;
+
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import com.google.protobuf.MessageLite;
+import com.google.protobuf.Parser;
+import java.io.IOException;
+import org.openyolo.protocol.MalformedDataException;
+
+/**
+ * Extracts protocol buffer messages byte array extras in Intents.
+ */
+public final class IntentProtocolBufferExtractor {
+
+    /**
+     * Attempts to extract a protocol buffer from the specified extra.
+     * @throws MalformedDataException if the intent is null, the extra is missing or not a byte
+     *     array, or the protocol buffer could not be parsed.
+     */
+    @NonNull
+    public static <T extends MessageLite> T extract(
+            @NonNull String extraName,
+            @NonNull Parser<T> protoParser,
+            @NonNull String failureDescription,
+            @Nullable Intent intent)
+            throws MalformedDataException {
+
+        if (intent == null) {
+            throw new MalformedDataException(failureDescription);
```

Comment:
Nit: We can throw more descriptive error messages crafting our own.
E.g.: null case: "Expected intent to contain non-null byte array for extra {extraName}"